flask_dance/consumer/oauth2.py

```diff
@@ -203,12 +214,25 @@ def teardown_session(self, exception=None):
     def login(self):
         log.debug("client_id = %s", self.client_id)
         self.session.redirect_uri = url_for(".authorized", _external=True)
+        code_verifier = None
+        if self._use_pkce:
+            code_verifier = create_code_verifier()
+            code_challenge = create_code_challenge(code_verifier, method=self.code_challenge_method)
+            self.authorization_url_params.update({
+                "code_challenge_method": self.code_challenge_method,
+                "code_challenge": code_challenge,
+            })
+
         url, state = self.session.authorization_url(
             self.authorization_url, state=self.state, **self.authorization_url_params
         )
         state_key = f"{self.name}_oauth_state"
         flask.session[state_key] = state
         log.debug("state = %s", state)
+        if code_verifier:
+            code_verifier_key = f"{self.name}_{state}_oauth_code_verifier"
```

Comment:
I don't think we need to include the `state` token in this key name. I would suggest:
```suggestion
            code_verifier_key = f"{self.name}_pkce_code_verifier"
```

Reply:
I was afraid of an edge case where more than one authentication process happens at once. 
I don't know if it's even a problem, since if the process is interrupted at the login stage and started again we will update the same key with the new `code_verifier` value which is the desired behavior.
I went for saving the `code_verifier` under the state aware key, but if you think that's an excessive solution, I don't have anything against simplifying it.

Reply:
There should not be more than one authentication process happening simultaneously, certainly not for the same OAuth provider. I'm not even sure the `self.name` reference is needed in the key name (for this or the `state` token), although it may help for debugging; adding a `state` reference is certainly excessive.